### PR TITLE
Fix SetAllowGetMethodPayload option ignored for io.Reader payload

### PR DESCRIPTION
**Pull request by picollomartin**
Hi 👋 !

We detect some inconsistent behaviour of the current implementation of the  `SetAllowGetMethodPayload` based on the input type passed to `SetBody` option:

- SetAllowGetMethodPayload = false
  - Body type = io.Reader => body is sent in GET
  - Body type = non io.Reader => body is not sent in GET
   
 - SetAllowGetMethodPayload = true
   - Body type = io.Reader => body is sent in GET
   - Body type = non io.Reader => body is sent in GET
   
Basically you can bypass the option AllowGetMethodPayload if you send an io.Reader as a Body, this fixes that for be consistent to all types and add a test case for test disabled option

### CODE REVIEW

**Comment by jeevatkm** [comment]
@picollomartin I'm sorry for the delayed attention on the PR. Thank you for your contribution. 
Agreed, inconsistent behavior will lead to consuming investigation time. This PR provided insight for #618.
